scanpy/get/_aggregated.py

```diff
@@ -0,0 +1,382 @@
+
+    def sum(self) -> Array:
+        """\
+        Compute the sum per feature per group of observations.
+
+        Returns
+        -------
+        Array of sum.
+        """
+        return utils.asarray(self.indicator_matrix @ self.data)
+
+    def mean(self) -> Array:
+        """\
+        Compute the mean per feature per group of observations.
+
+        Returns
+        -------
+        Array of mean.
+        """
+        return (
+            utils.asarray(self.indicator_matrix @ self.data)
+            / np.bincount(self.groupby.codes)[:, None]
+        )
+
+    def mean_var(self, dof: int = 1) -> tuple[np.ndarray, np.ndarray]:
+        """\
+        Compute the count, as well as mean and variance per feature, per group of observations.
+
+        The formula `Var(X) = E(X^2) - E(X)^2` suffers loss of precision when the variance is a
+        very small fraction of the squared mean. In particular, when X is constant, the formula may
+        nonetheless be non-zero. By default, our implementation resets the variance to exactly zero
+        when the computed variance, relative to the squared mean, nears limit of precision of the
+        floating-point significand.
+
+        Params
+        ------
+        dof
+            Degrees of freedom for variance.
+
+        Returns
+        -------
+        Object with `count`, `mean`, and `var` attributes.
+        """
+        assert dof >= 0
+
+        group_counts = np.bincount(self.groupby.codes)
+        mean_ = self.mean()
+        # sparse matrices do not support ** for elementwise power.
+        mean_sq = (
+            utils.asarray(self.indicator_matrix @ _power(self.data, 2))
+            / group_counts[:, None]
+        )
+        sq_mean = mean_**2
+        var_ = mean_sq - sq_mean
+        # TODO: Why these values exactly? Because they are high relative to the datatype?
+        # (unchanged from original code: https://github.com/scverse/anndata/pull/564)
+        precision = 2 << (42 if self.data.dtype == np.float64 else 20)
+        # detects loss of precision in mean_sq - sq_mean, which suggests variance is 0
+        var_[precision * var_ < sq_mean] = 0
+        if dof != 0:
+            var_ *= (group_counts / (group_counts - dof))[:, np.newaxis]
+        return mean_, var_
+
+
+def _power(X: Array, power: float | int) -> Array:
+    """\
+    Generate elementwise power of a matrix.
+
+    Needed for non-square sparse matrices because they do not support ** so the `.power` function is used.
+
+    Params
+    ------
+    X
+        Matrix whose power is to be raised.
+    power
+        Integer power value
+
+    Returns
+    -------
+    Matrix whose power has been raised.
+    """
+    return X**power if isinstance(X, np.ndarray) else X.power(power)
+
+
+@singledispatch
+def aggregate(
+    adata: AnnData,
+    by: str | list[str],
+    func: AggType | Iterable[AggType],
+    *,
+    axis: Literal[0, 1] | None = None,
```

Comment:
Since we have a consensus now:

```suggestion
    axis: Literal["obs", 0, "var", 1] | None = None,
```